### ISSUE TITLE
Color translation wrong in mono?

### ISSUE THREAD

**Issue by carlosbispo** [godotengine/godot#40075]
**Godot version: 3.2.2 Mono**

**OS : Windows 10**

**Issue description:**
When trying to use Color constructor in Mono with texturerect, wrong color appears, in GDNative everything is fine
GD Native
draw_rect(Rect2(0, 0, 10, 10), Color(0xff0000ff)) -> Red, as expected

C#
DrawRect(new Rect2(0, 0, 10, 10), new Color(0xff0000ff)) -> Transparent Color?

**Comment by Calinou** [comment]
Try writing the color using `Color(R, G, B[, A])` notation instead, which is less error-prone.

**Comment by akien-mga** [comment]
The Mono bindings probably got the `Color::hex` and `Color::hex64` backwards. 

It's RGBA, and thus parsed from right to left ARGB:
https://github.com/godotengine/godot/blob/aa564739dde144687ad71dae853cc3aa4d49153f/core/color.cpp#L226-L236

**Comment by Calinou** [comment]
> It's RGBA, and thus parsed from right to left ARGB:

See also https://github.com/godotengine/godot/pull/36992 :wink:

**Comment by akien-mga** [comment]
That was a typo, it's ABGR (RGBA from right to left). It's not linked to the use of ARGB for HTML notation.

**Comment by akien-mga** [comment]
The Mono `Color(int)` constructor seems correct though:
```csharp
        public Color(uint rgba)
        {
            a = (rgba & 0xFF) / 255.0f;
            rgba >>= 8;
            b = (rgba & 0xFF) / 255.0f;
            rgba >>= 8;
            g = (rgba & 0xFF) / 255.0f;
            rgba >>= 8;
            r = (rgba & 0xFF) / 255.0f;
        }
```

**Comment by akien-mga** [comment]
I can confirm the bug in 3.2.2-stable.

In GDScript:
```
print(Color(0xff0000ff))  # 1,0,0,1
```
In C#:
```
GD.Print(new Color(0xff0000ff))  # 0,0,0.9961089,0.003891051
```

**Comment by akien-mga** [comment]
It seems fixed in the `master` branch.

**Comment by akien-mga** [comment]
> It seems fixed in the `master` branch.

That might be related to #39609 which was fixed in #39629.

**Comment by aaronfranke** [comment]
The problem is that this value is interpreted as a `long`. You can see this by hovering in an IDE:

![Screenshot from 2020-07-03 20-59-44](https://user-images.githubusercontent.com/1646875/86502424-edb0df80-bd70-11ea-8145-f1f451d3ebe2.png)

This code works correctly:

```cs
GD.Print(new Color(Colors.Red.ToRgba32())); // Prints 1,0,0,1
```

This was fixed by #35649 for 4.0, but you have to explicitly write it unsigned: 

```cs
GD.Print(new Color((uint)0xFF0000FF));
```

In the 3.2 branch you have to do this: 

```cs
GD.Print(new Color(unchecked((int)0xFF0000FF)));
```

But ideally, you would just do this instead, only using the integer constructors for stored values:

```cs
GD.Print(new Color(1, 0, 0, 1));
```

**Comment by akien-mga** [comment]
> This was fixed by #35649 for 4.0, but you have to explicitly write it unsigned:
> 
> ```cs
> GD.Print(new Color((uint)0xFF0000FF));
> ```

When I tested it worked fine in `master` with `GD.Print(new Color(0xFF0000FF));`.
